packages/@aws-cdk/cloudformation-include/lib/cfn-include.ts

```diff
@@ -152,6 +175,32 @@ export class CfnInclude extends core.CfnElement {
     this.parameters[logicalId] = cfnParameter;
   }
 
+  private createOutput(logicalId: string): void {
+    const self = this;
+    const outputAttributes = new cfn_parse.CfnParser({
+      finder: {
+        findResource(lId): core.CfnResource | undefined {
+          return self.resources[lId];
+        },
+        findRefTarget(elementName: string): core.CfnElement | undefined {
+          return self.resources[elementName] ?? self.parameters[elementName];
+        },
+        findCondition(): undefined {
+          return undefined;
+        },
+      },
+    }).parseValue(this.template.Outputs[logicalId]);
+    const cfnOutput = new core.CfnOutput(this, logicalId, {
+      value: outputAttributes.Value,
+      description: outputAttributes.Description,
+      exportName: outputAttributes.Export,
+      condition: outputAttributes.Condition ? self.getCondition(outputAttributes.Condition) : undefined,
```

Comment:
Doesn't outputAttributes.Condition being undefined here mean that the template didn't specify a Condition as a member of the Output object?

Reply:
Did you mean the findCondition() member of the finder variable?

Reply:
I meant the error message you would get if the Output referenced a Condition that is not in the template.

Reply:
something like this, but with better formatting?

`      condition: (() => {
        try { return self.getCondition(outputAttributes.Condition); }
        catch (e) {
          throw new Error(`Output with name '${logicalId}' refers to a Condition with name\
 '${outputAttributes.Condition}' which was not found in this template`);
        }
      })()`

Reply:
Yes, although I don't love that, because it swallows all exceptions, not only the one that you want here.

How about just checking the `conditions` field in this class?

Reply:
Like this?

`        if (this.conditions[outputAttributes.Export])
        {
          return outputAttributes.Export.Name;
        }

        throw new Error(`Output with name '${logicalId}' refers to a Condition with name\
 '${outputAttributes.Condition}' which was not found in this template`);
      })()`

Reply:
Something like that. I probably wouldn't do it in an immediately-invoked function expression, but just have a local variable like `outputCondition`.